File: scalang/src/main/scala/com/kapilkoju/autopos/trade/purchase/service/PurchaseService.scala
package com.kapilkoju.autopos.trade.purchase.service

import com.kapilkoju.autopos.trade.purchase.domain.{Purchase, PurchaseLine}
import org.springframework.data.domain.{Page, Pageable}

trait PurchaseService {

  def getPurchases(pageable: Pageable): Page[Purchase]

  def createNewPurchase(): Purchase

  def updatePurchase(purchase: Purchase): Purchase

  def deletePurchase(purchase: Purchase): Unit

  def addPurchaseLine(purchase: Purchase, purchaseLine: PurchaseLine): PurchaseLine

  def updatePurchaseLine(purchase: Purchase, purchaseLine: PurchaseLine): PurchaseLine

  def deletePurchaseLine(purchaseLine: PurchaseLine): Unit

  def getPurchaseLines(itemId: Long): List[PurchaseLine]

}

File: src/main/kotlin/com/kapilkoju/autopos/trade/purchase/service/PurchaseService.kt
package com.kapilkoju.autopos.trade.purchase.service

import com.kapilkoju.autopos.trade.purchase.domain.Purchase
import com.kapilkoju.autopos.trade.purchase.domain.PurchaseLine
import com.kapilkoju.autopos.trade.purchase.dto.CreatePurchaseDto
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface PurchaseService {

    fun getPurchases(vat: Boolean, pageable: Pageable): Page<Purchase>

    fun createNewPurchase(createPurchaseDto: CreatePurchaseDto): Purchase

    fun updatePurchase(purchase: Purchase): Purchase

    fun deletePurchase(purchase: Purchase)

    fun addPurchaseLine(purchase: Purchase, purchaseLine: PurchaseLine): PurchaseLine

    fun updatePurchaseLine(purchase: Purchase, purchaseLine: PurchaseLine): PurchaseLine

    fun deletePurchaseLine(purchaseLine: PurchaseLine)

    fun getPurchaseLines(itemId: Long): List<PurchaseLine>

}
